src/Orleans.Core/Configuration/OptionLogger/IOptionsLogger.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Orleans
+{ 
+    internal class ClientOptionsLogger : OptionsLogger, ILifecycleParticipant<IClusterClientLifecycle>
+    {
+        private int ClientOptionLoggerLifeCycleRing = 1000;
+        public ClientOptionsLogger(ILogger<ClientOptionsLogger> logger, IServiceProvider services)
+            : base(logger, services)
+        {
+        }
+
+        public void Participate(IClusterClientLifecycle lifecycle)
+        {
+            lifecycle.Subscribe(ClientOptionLoggerLifeCycleRing, this.OnRuntimeInitializeStart);
+        }
+
+        public Task OnRuntimeInitializeStart(CancellationToken token)
```

Comment:
This name doesn't quite make sense because client life cycle doesn't have a lifecycle ring enum.. so I was assuming there will be a RuntimeInit stage. Hence this method is called OnRuntimeInit because it is going to subscribe to that ring. I was just using naming for life participant from other code samples. 

But seems to you using a int.Minvalue is good?

Reply:
This name was chosen in other code because that was when the logic was run..  As for when to run this, yeah, I favor Int.Min, but I don't have a strong opinion of this.  I think the goal is for this to run before most other things in the system.

Reply:
see my comments on my confusion on lifecycle ring above... I don't think we should have random int lying around the code base

Reply:
Compromise:  How about we add the following stages to the SiloLifecycle stages
        public const int First = Int.Min;
        public const int Last = Int.Max;

Reply:
added 
public const int First = Int.Min; to silo lifecycle stages